apps/openmw/mwworld/worldimp.cpp

```diff
@@ -2753,44 +2732,43 @@ namespace MWWorld
             return lhs->getDestCell() < rhs->getDestCell();
         });
 
-        for (std::vector<const MWWorld::CellRef *>::const_iterator it = sortedDoors.begin(); it != sortedDoors.end(); ++it)
+        for (const MWWorld::CellRef* door : sortedDoors)
         {
-            MWWorld::CellStore *source = 0;
+            MWWorld::CellStore *source = nullptr;
 
             // door to exterior
-            if ((*it)->getDestCell().empty())
+            if (door->getDestCell().empty())
             {
                 int x, y;
-                ESM::Position doorDest = (*it)->getDoorDest();
+                ESM::Position doorDest = door->getDoorDest();
                 positionToIndex(doorDest.pos[0], doorDest.pos[1], x, y);
                 source = getExterior(x, y);
             }
             // door to interior
             else
             {
-                source = getInterior((*it)->getDestCell());
+                source = getInterior(door->getDestCell());
             }
-            if (0 != source)
+            if (source)
             {
                 // Find door leading to our current teleport door
                 // and use its destination to position inside cell.
-                const DoorList &destinationDoors = source->getReadOnlyDoors().mList;
-                for (DoorList::const_iterator jt = destinationDoors.begin(); jt != destinationDoors.end(); ++jt)
+                for (const MWWorld::LiveCellRef<ESM::Door>& destDoor : source->getReadOnlyDoors().mList)
                 {
-                    if ((*it)->getTeleport() &&
```

Comment:
Is the getTeleport() removal intended?

Reply:
Yes, there can't be non-teleport doors in the sortedDoors vector because non-teleport doors are skipped when the vector is created.

Reply:
Ok.